### PR TITLE
Add unit tests for bridge.go

### PR DESCRIPTION
**Pull request by PabloHiro**
## Summary
Adds comprehensive unit tests for `pkg/utils/bridge.go` to improve test coverage and code quality. Also improves variable naming for better code readability.

### 1. Improved Code Clarity
Renamed variables in `bridge.go` for better readability:
- `c1`/`c2` → `sourceConnection`/`destinationConnection`
- `c1Name`/`c2Name` → `connection1Name`/`connection2Name`

### 2. Comprehensive Unit Tests
- **100% code coverage** for bridge.go
- 9 test functions covering all code paths and edge cases
- Uses existing gomock mocks from `pkg/utils/mock_utils`
- Thread-safe implementation with mutex protection for shared buffers
- Timeout protection for all tests (2-second timeout)

## Test Cases
This PR implements tests for:
- ✅ Successful bidirectional data bridging
- ✅ Empty data handling (immediate EOF)
- ✅ Read error handling
- ✅ Write error handling
- ✅ Partial write detection and handling
- ✅ Large data transfers (> NormalBufferSize)
- ✅ Unidirectional data flow
- ✅ Concurrent bidirectional data flow
- ✅ NormalBufferSize constant verification

## Test Results
All tests pass successfully:
```
PASS
ok      github.com/ansible/receptor/pkg/utils  0.008s
coverage: 100.0% for bridge.go
```

## Motivation
This addresses the need for better test coverage identified in the SonarCloud analysis. The `bridge.go` file had no test coverage previously, and these tests bring it to 100%.

## Test Implementation
- Uses existing gomock mocks from `pkg/utils/mock_utils` for consistency
- Thread-safe mock implementation with mutex protection
- Timeout protection for all tests to prevent hanging
- Tests for both normal and error conditions
- Uses DoAndReturn for custom behavior in mock expectations

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: pkg/utils/bridge.go [hunk 1]
```diff
@@ -11,33 +11,33 @@ import (
 const NormalBufferSize = 65536
 
 // BridgeConns bridges two connections, like netcat.
-func BridgeConns(c1 io.ReadWriteCloser, c1Name string, c2 io.ReadWriteCloser, c2Name string, logger *logger.ReceptorLogger) {
+func BridgeConns(conn1 io.ReadWriteCloser, conn1Name string, conn2 io.ReadWriteCloser, conn2Name string, logger *logger.ReceptorLogger) {
 	doneChan := make(chan bool)
-	go bridgeHalf(c1, c1Name, c2, c2Name, doneChan, logger)
-	go bridgeHalf(c2, c2Name, c1, c1Name, doneChan, logger)
+	go bridgeHalf(conn1, conn1Name, conn2, conn2Name, doneChan, logger)
+	go bridgeHalf(conn2, conn2Name, conn1, conn1Name, doneChan, logger)
```
Comment (tyraziel) [inline pkg/utils/bridge.go]: Expand this out to connection over conn.

File: pkg/utils/bridge.go [hunk 2]
```diff
@@ -11,33 +11,33 @@ import (
...
 	<-doneChan
 	<-doneChan
 }
 
-// BridgeHalf bridges the read side of c1 to the write side of c2.
-func bridgeHalf(c1 io.ReadWriteCloser, c1Name string, c2 io.ReadWriteCloser, c2Name string, done chan bool, logger *logger.ReceptorLogger) {
-	logger.Trace("    Bridging %s to %s\n", c1Name, c2Name)
+// BridgeHalf bridges the read side of conn1 to the write side of conn2.
+func bridgeHalf(conn1 io.ReadWriteCloser, conn1Name string, conn2 io.ReadWriteCloser, conn2Name string, done chan bool, logger *logger.ReceptorLogger) {
+	logger.Trace("    Bridging %s to %s\n", conn1Name, conn2Name)
```
Comment (tyraziel) [inline pkg/utils/bridge.go]: As reviewed:

conn1 -> sourceConnection
conn2 -> destinationConnection

**Review by tyraziel — CHANGES_REQUESTED** [review]

File: pkg/utils/bridge_test.go [hunk 3]
```diff
@@ -0,0 +1,445 @@
...
+		// Success
+	case <-time.After(2 * time.Second):
+		t.Fatal("BridgeConns did not complete within timeout")
+	}
+
+	// Verify data was bridged correctly
+	mu.Lock()
+	conn1WrittenData := conn1Written.Bytes()
+	conn2WrittenData := conn2Written.Bytes()
+	mu.Unlock()
```
Comment (davemulford) [inline pkg/utils/bridge_test.go]: Do we need a mutex to lock here since the go routine has already completed earlier?

File: pkg/utils/bridge_test.go [hunk 4]
```diff
@@ -0,0 +1,445 @@
...
+		return reader1.Read(p)
+	}).AnyTimes()
+	conn1.EXPECT().Close().Return(nil).AnyTimes()
+
+	// c2 returns error on write
+	conn2.EXPECT().Read(gomock.Any()).Return(0, io.EOF).AnyTimes()
+	conn2.EXPECT().Write(gomock.Any()).Return(0, errors.New("write error")).AnyTimes()
+	conn2.EXPECT().Close().Return(nil).AnyTimes()
+
+	log := logger.NewReceptorLogger("")
```
Comment (davemulford) [inline pkg/utils/bridge_test.go:189]: We could look into what was logged and make sure the errors we expect are there.

File: pkg/utils/bridge_test.go [hunk 5]
```diff
@@ -0,0 +1,445 @@
...
+	conn2.EXPECT().Write(gomock.Any()).DoAndReturn(func(p []byte) (int, error) {
+		if len(p) > 1 {
+			return len(p) / 2, nil // Partial write
+		}
+
+		return len(p), nil
+	}).AnyTimes()
+	conn2.EXPECT().Close().Return(nil).AnyTimes()
+
+	log := logger.NewReceptorLogger("")
```
Comment (davemulford) [inline pkg/utils/bridge_test.go:290]: Check that the log has emitted what we would expect for a partial write.

File: pkg/utils/bridge_test.go [hunk 6]
```diff
@@ -0,0 +1,445 @@
...
+	reader1 := bytes.NewReader(data)
+	conn1.EXPECT().Read(gomock.Any()).DoAndReturn(func(p []byte) (int, error) {
+		return reader1.Read(p)
+	}).AnyTimes()
+	conn1.EXPECT().Write(gomock.Any()).DoAndReturn(func(p []byte) (int, error) {
+		mu.Lock()
+		defer mu.Unlock()
+
+		return conn1Written.Write(p)
+	}).AnyTimes()
```
Comment (PabloHiro) [inline pkg/utils/bridge_test.go]: Is this EXPECT needed?

**Review by davemulford — CHANGES_REQUESTED** [review]
Looks great! Just a couple of questions about mutex locking and whether we can check the logs to ensure expected behavior.

File: pkg/utils/bridge.go [hunk 7]
```diff
@@ -48,8 +49,10 @@ func bridgeHalf(c1 io.ReadWriteCloser, c1Name string, c2 io.ReadWriteCloser, c2N
 			}
 		}
 		if shouldClose {
-			logger.Trace("    Stopping bridge %s to %s\n", c1Name, c2Name)
-			_ = c2.Close()
+			logger.Trace("    Stopping bridge %s to %s\n", sourceConnectionName, destinationConnectionName)
+			if err := destinationConnection.Close(); err != nil {
```
Comment (PabloHiro) [inline pkg/utils/bridge.go:53]: Functional change:
* Error is now logged

File: pkg/utils/bridge.go [hunk 8]
```diff
@@ -11,33 +12,33 @@ import (
...
 		done <- true
 	}()
 	buf := make([]byte, NormalBufferSize)
 	shouldClose := false
 	for {
-		n, err := c1.Read(buf)
+		n, err := sourceConnection.Read(buf)
 		if err != nil {
-			if err.Error() != "EOF" && !strings.Contains(err.Error(), "use of closed network connection") {
+			if !errors.Is(err, io.EOF) && !errors.Is(err, net.ErrClosed) {
```
Comment (PabloHiro) [inline pkg/utils/bridge.go:34]: This should be equivalent

File: pkg/utils/bridge_test.go (hunk 4 above)
Comment (PabloHiro) [reply pkg/utils/bridge_test.go:189]: Done!

File: pkg/utils/bridge_test.go (hunk 5 above)
Comment (PabloHiro) [reply pkg/utils/bridge_test.go:290]: Done!

File: pkg/utils/bridge_test.go (hunk 3 above)
Comment (PabloHiro) [reply pkg/utils/bridge_test.go]: Not needed indeed, removed!

File: pkg/utils/bridge_test.go (hunk 6 above)
Comment (PabloHiro) [reply pkg/utils/bridge_test.go]: It was not needed

File: pkg/utils/bridge.go (hunk 2 above)
Comment (PabloHiro) [reply pkg/utils/bridge.go]: Renamed :+1:

**Review by davemulford — APPROVED** [review]
LGTM!

**Comment by PabloHiro** [comment]
This adds cognitive complexity to the method, I will address this in follow-up PRs to keep the diff readable in this one.

**Review by tyraziel — APPROVED** [review]
Thanks for making the variable name changes!

LGTM.
